services/platform/platform.go

```diff
@@ -174,37 +233,67 @@ func (s *Service) connect(ctx context.Context, params *connectPMMParams) (*ssoDe
 	})
 	if err != nil {
 		s.l.Errorf("Failed to marshal request data: %s", err)
-		return nil, status.Error(codes.Internal, "Internal server error")
+		return nil, status.Error(codes.Internal, internalServerError)
 	}
 
-	client := http.Client{Timeout: s.platformAPITimeout}
 	req, err := http.NewRequestWithContext(ctx, http.MethodPost, endpoint, bytes.NewReader(marshaled))
 	if err != nil {
 		s.l.Errorf("Failed to build Connect to Platform request: %s", err)
-		return nil, status.Error(codes.Internal, "Internal server error")
+		return nil, status.Error(codes.Internal, internalServerError)
 	}
 	req.SetBasicAuth(params.email, params.password)
-	resp, err := client.Do(req)
+	resp, err := s.client.Do(req)
 	if err != nil {
 		s.l.Errorf("Connect to Platform request failed: %s", err)
-		return nil, status.Error(codes.Internal, "Internal server error")
+		return nil, status.Error(codes.Internal, internalServerError)
 	}
 	defer resp.Body.Close() //nolint:errcheck
 
 	decoder := json.NewDecoder(resp.Body)
 	if resp.StatusCode != http.StatusOK {
 		var gwErr grpcGatewayError
 		if err := decoder.Decode(&gwErr); err != nil {
-			s.l.Errorf("Connect to Platform request failed and we faild to decode error message: %s", err)
-			return nil, status.Error(codes.Internal, "Internal server error")
+			s.l.Errorf("Connect to Platform request failed and we failed to decode error message: %s", err)
+			return nil, status.Error(codes.Internal, internalServerError)
 		}
 		return nil, status.Error(codes.Code(gwErr.Code), gwErr.Message)
 	}
 
 	var response connectPMMResponse
 	if err := decoder.Decode(&response); err != nil {
 		s.l.Errorf("Failed to decode response into SSO details: %s", err)
-		return nil, status.Error(codes.Internal, "Internal server error")
+		return nil, status.Error(codes.Internal, internalServerError)
 	}
 	return response.SSODetails, nil
 }
+
+func (s *Service) disconnect(ctx context.Context, params *disconnectPMMParams) error {
+	endpoint := fmt.Sprintf("https://%s/v1/orgs/inventory/%s:disconnect", s.host, params.PMMServerID)
+	req, err := http.NewRequestWithContext(ctx, http.MethodDelete, endpoint, nil)
```

Comment:
Since this API method updated to be custom, it uses POST instead of DELETE

Reply:
Updated.